model_compression_toolkit/data_generation/common/data_generation.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+# Import required modules and classes
+import time
+from typing import Callable, Any
+
+from charset_normalizer.md import List
+
+from model_compression_toolkit.data_generation.common.data_generation_config import DataGenerationConfig
+from model_compression_toolkit.data_generation.common.model_info_exctractors import ActivationExtractor, \
+    OrigBNStatsHolder
+from model_compression_toolkit.data_generation.common.optimization_utils import AllImagesOptimizationHandler
+from model_compression_toolkit.logger import Logger
+
+
+def data_generation(
+        data_generation_config: DataGenerationConfig,
+        activation_extractor: ActivationExtractor,
+        orig_bn_stats_holder: OrigBNStatsHolder,
+        all_imgs_opt_handler: AllImagesOptimizationHandler,
+        image_pipeline: Callable,
```

Comment:
Should the type of image_pipeline be BaseImagePipeline? (later in this method you're expecting it to have methods like "image_input_manipulation", which is part of the BaseImagePipeline class).

Reply:
nice catch!